packages/entity-list/src/components/Table/Table.js

```diff
@@ -1,141 +1,144 @@
-    if (props.onSelectChange) {
-      props.onSelectChange(keys, isSelected)
-    }
-  }
+  useDeepCompareEffect(() => {
+    const c = columnEnhancers.reduce(
+      (acc, columnEnhancer) => columnEnhancer.shouldApply(props)
+        ? columnEnhancer.apply(acc, {singleSelectHandler, multiSelectHandler, isSelected})
+        : acc,
+      props.columnDefinitions
+    )
+    setColumns(c)
+  }, [props.columnDefinitions, props.tableSelectionStyle])
 
-  const handleSelectionChange = (row, isSelected) => {
-    if (props.onSelectChange) {
-      props.onSelectChange([row.__key], isSelected)
-    }
-  }
+  const showPagination = props.entityCount - props.limit > 0
 
-  const getSelectionMode = () =>
-    (props.tableSelectionStyle && props.tableSelectionStyle !== selectionStyles.NONE)
-      ? props.tableSelectionStyle === selectionStyles.SINGLE
-        ? {mode: 'radio'}
-        : {mode: 'checkbox'}
-      : {}
-
-  const selectRow = {
-    ...(getSelectionMode()),
-    onSelect: handleSelectionChange,
-    onSelectAll: handleAllSelectionChange,
-    selected: props.selection ? props.selection : [],
-    className: 'selected'
+  const thOnClick = column => e => {
+    if (!resizingColumn && column.sortable) {
+      props.setSortingInteractive(column.id, e.shiftKey)
+    }
   }
 
-  const tableOption = {
-    noDataText: props.inProgress
-      ? msg('client.entity-list.dataLoading')
-      : msg('client.entity-list.noData'),
-    onPageChange: onPageChange,
-    onRowClick: handleRowClick,
-    onSortChange: onSortChange,
-    page: props.currentPage,
-    paginationPanel: PaginationPanel,
-    paginationShowsTotal: renderShowsTotal,
-    sizePerPage: props.limit,
-    ...sortingOptions()
+  const trOnClick = entity => e => {
+    if (e.shiftKey) {
+      singleSelectHandler(entity.__key, true, true)
+    } else if (e.metaKey || e.ctrlKey) {
+      singleSelectHandler(entity.__key)
+    } else {
+      props.onRowClick(entity.__key)
+    }
   }
 
-  const showPagination = props.entityCount - props.limit > 0 && !props.inProgress
-
-  const cellFormatter = (column, idx) => (cell, entity) => (
-    <span key={idx} data-cy="list-cell">
-      {column.children.map(child => cellRenderer(child, entity, props.parent, {refresh: props.refresh}, props.intl))}
-    </span>
-  )
-
-  const getLinkColumn = () =>
-    props.linkFactory && props.linkFactory.detail && props.showLink
-      ? <TableHeaderColumn
-        width="30px"
-        dataFormat={(cell, row, formatExtraData, rowIdx) => (
-          <span onClick={e => e.stopPropagation()}>
-            {props.linkFactory.detail(null, null, row.__key, <Icon icon="arrow-right"/>)}
-          </span>
-        )}
+  const ThContent = ({column}) =>
+    column.headerRender
+      ? column.headerRender({...props, singleSelectHandler, multiSelectHandler, isSelected})
+      : column.label
+
+  const InProgressRow = () => <tr><td className="fullRow progress">
+    <LoadingSpinner size="20"/>
```

Comment:
(I know it's not a new component, but why is that size prop a string and not a number?)

Reply:
because it is set as css attribute fontSize and therefore could have a unit. e.g. `2rem`